Docker/Dockerfile.main

```diff
@@ -2,37 +2,40 @@
 # application based on the official python image and the application code
 
 # Stage 1: Build the application
-FROM python:3.10 as build
+FROM python:3.10.15 as build
 LABEL Stage="Build"
 
 ENV HOME_DIR=/usr/src/backend
+ENV DATA_DIR=/data
 ENV PYTHONUNBUFFERED 1
 
 # Set the working directory to /usr/src/backend
 WORKDIR ${HOME_DIR}
 
+RUN mkdir -p ${DATA_DIR}
+
 COPY requirements.txt /requirements.txt
 
 # install uv. This makes installation of packages faster compared to pip
 ADD https://astral.sh/uv/install.sh /install.sh
 
 #Install system dependencies. git is needed because there are some packages that are not available in PyPI
-RUN apt-get update && apt-get -y install libpq-dev gcc wget curl git \
+RUN apt-get update && apt-get -y install libpq-dev gcc wget curl git curl \
     && chmod -R 755 /install.sh && /install.sh && rm /install.sh \
     && set -ex apt-get autoremove -y --purge wget && rm -rf /var/lib/apt/lists/* \
     && /root/.cargo/bin/uv pip install --system --upgrade pip \
     && /root/.cargo/bin/uv pip install --system --no-cache-dir -r /requirements.txt \
     && rm -rf /var/lib/apt/lists/* \
-    && apt-get clean 
-
-
+    && apt-get clean
 
-RUN mkdir -p ${HOME_DIR}/secrets \
-    && mkdir -p ${HOME_DIR}/dbt/dbt \
-    && mkdir -p ${HOME_DIR}/ddpui/logs 
+RUN mkdir -p ${DATA_DIR}/secrets \
```

Comment:
you need to create the `dbt` folder because you're going to `pip install` things in the next step, but you don't need to create `secrets` or `clients_dbt`. Docker will create them during the volume-mapping stage of the docker-compose

chat gpt said:
```
In Docker Compose, when you define a volume mapping like - host_path:/path/in/container, the container path (/path/in/container) does not need to exist in the image. Docker will create the entire directory structure in the container if it doesn't already exist, based on the structure specified in the volume mapping.
```

Reply:
Agreed. I tested it out, it does create the directories.